org.adempiere.base/src/org/compiere/util/Login.java

```diff
@@ -1252,15 +1252,29 @@ public KeyNamePair[] getClients(String app_user, String app_pwd) {
 		return getClients(app_user, app_pwd, null);
 	}
 
+	/**
+	 * Validate Client Login. Sets Context with login info
+	 * 
+	 * @param app_user  user id
+	 * @param app_pwd   password
+	 * @param roleTypes comma separated list of the role types allowed to login
+	 *                  (NULL can be added)
+	 * @return client array or null if in error.
+	 */
+	public KeyNamePair[] getClients(String app_user, String app_pwd, String roleTypes) {
+		return getClients(app_user, app_pwd, null, false);
+	}
+
 	/**
 	 *  Validate Client Login.
 	 *  Sets Context with login info
 	 *  @param app_user user id
 	 *  @param app_pwd password
 	 *  @param roleTypes comma separated list of the role types allowed to login (NULL can be added)
+	 *  @param isSSOLogin check only base on user Name
 	 *  @return client array or null if in error.
 	 */
-	public KeyNamePair[] getClients(String app_user, String app_pwd, String roleTypes) {
+	public KeyNamePair[] getClients(String app_user, String app_pwd, String roleTypes, boolean isSSOLogin) {
```

Comment:
With isSSOLogin=true, it basically remove all validation. Since this is a public method, it is probably more secure if isSSOLogin is replace with the SSO token that we can verify the validity of it using the SSO Principal provider.

Reply:
This is still pending.